packages/appium/lib/config-file.js

```diff
@@ -0,0 +1,293 @@
+        // enables use to use `"type": ["foo", "bar"]` in schema
+        allowUnionTypes: true,
+        // enables us to use custom properties (e.g., `appiumDest`)
+        strict: false,
+      }),
+    );
+    return ajv.compile(schema);
+  },
+);
+
+/**
+ * Validates a raw config object against the config schema.
+ * Initializes the Ajv singleton, if not already done.
+ * @param {object} [config] - Configuration object to validate. Could be anything.
+ * @returns {import('ajv').ErrorObject[]} Array of error objects from Ajv, if any.
+ */
+function validateConfigFile (config = {}) {
+  const validate = validator();
+  // validate.errors will be non-empty if `validate()` returns `false`.
+  // ...yes, that is a weird API.
+  return validate(config) ? [] : [...validate.errors];
+}
+
+/**
+ * Given an optional path, read a config file.
+ * @param {string} [filepath] - Path to config file, if we have one
+ * @param {ReadConfigFileOptions} [opts] - Options
+ * @returns {Promise<ReadConfigFileResult>} Contains config and filepath, if found, and any errors
+ */
+export async function readConfigFile (filepath, opts = {}) {
+  const result = (await findConfigFile(filepath)) ?? {};
+  if (result.config && !result.isEmpty) {
+    log.debug(`Config file found at ${result.filepath}`);
+    const {normalizeKeys = true, pretty = true} = opts;
+    /** @type {ReadConfigFileResult} */
+    let retval;
+    const errors = validateConfigFile(result.config);
+    if (errors.length) {
+      /** @type {string} */
+      let json;
+      try {
+        json = await fs.readFile(result.filepath, 'utf8');
+      } catch {}
+      const reason = betterAjvErrors(schema, result.config, errors, {
+        json,
+        format: pretty ? 'cli' : 'js',
+      });
+      retval = reason ? {...result, errors, reason} : {...result, errors};
+    } else {
+      retval = {...result, errors};
+    }
+    if (normalizeKeys) {
+      // XXX blah
+      retval.config = objectKeysToCamelCase(retval.config);
+      ['sever', 'plugin', 'driver'].forEach((key) => {
+        if (retval.config[key]) {
+          retval.config[key] = objectKeysToCamelCase(retval.config[key]);
+        }
+      });
+    }
+    log.debug('Final config result: ', retval);
+    return retval;
+  }
+  return result;
+}
+
+/**
+ * Convert object key names from kebab-case to camelCase.
+ * @param {object} [obj]
+ * @returns New object with camelCased keys.
+ */
+function objectKeysToCamelCase (obj = {}) {
+  return _.mapKeys(obj, (value, arg) => _.camelCase(arg));
+}
+
+/**
+ * Converts flag aliases to actual flags.
+ * Any flag shorter than 3 characters is prefixed with a single dash, otherwise two.
+ * @param {string} alias - A flag alias (e.g. `verbose`) to convert to a flag (e.g., `--verbose`)
+ * @throws {TypeError} If `alias` is falsy
+ */
+function aliasToFlag (alias) {
+  if (!alias) {
+    throw new TypeError('falsy alias value');
+  }
+  return alias.length < 3 ? `-${alias}` : `--${alias}`;
```

Comment:
magic number here?

Reply:
I mean... yes, sort of.  all of our aliases less than length 3 are usable with `-`, and anything longer is `--`.